### PR TITLE
sse2neon 1.5.1 (new formula)

### PR DESCRIPTION
**Pull request by alexreg**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by alexreg** [comment]
Test should pass now.

**Comment by chenrui333** [comment]
```
  test.c:6:10: error: implicit declaration of function 'vaddd_s64' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
```

**Comment by alexreg** [comment]
@chenrui333 Weird that that's not present on Intel architectures... anyway, fixed.

**Review by chenrui333 — DISMISSED** [review]

File: Formula/sse2neon.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  head "https://github.com/DLTcollab/sse2neon.git", branch: "master"
+
+  def install
+    include.install "sse2neon.h"
+  end
+
+  test do
+    (testpath/"test.c").write <<~EOS
+      #include <assert.h>
+      #include <#{Hardware::CPU.intel? ? "emmintrin.h" : "sse2neon.h"}>
```
Comment (SMillerDev) [inline Formula/sse2neon.rb:16]: This means it won't test the one installed file on intel though

**Review by SMillerDev — CHANGES_REQUESTED** [review]

File: Formula/sse2neon.rb (hunk 1 above)
Comment (alexreg) [reply Formula/sse2neon.rb:16]: @SMillerDev It can't do, because sse2neon simply cannot be used on Intel.

Comment (SMillerDev) [reply Formula/sse2neon.rb:16]: In that case it needs a `depends_on :arm64`

Comment (alexreg) [reply Formula/sse2neon.rb:16]: Ah okay. I was looking for something like that initially, but it seems it isn't yet used in any homebrew-core formula! Thanks.

**Comment by alexreg** [comment]
Test failure

```
Error: sse2neon: Unsupported special dependency :arm64
```

File: Formula/sse2neon.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/sse2neon.rb]: Nope, this might be the very first ARM only formula

Comment (alexreg) [reply Formula/sse2neon.rb]: @SMillerDev Hah, fair enough. Does the test failure mean anything to you? How do we fix that?

File: Formula/sse2neon.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+class Sse2neon < Formula
+  desc "Translator from Intel SSE intrinsics to Arm/Aarch64 NEON implementation"
+  homepage "https://github.com/DLTcollab/sse2neon"
+  url "https://github.com/DLTcollab/sse2neon/archive/v1.5.1.tar.gz"
+  sha256 "4001e2dfb14fcf3831211581ed83bcc83cf6a3a69f638dcbaa899044a351bb2a"
+  license "MIT"
+  head "https://github.com/DLTcollab/sse2neon.git", branch: "master"
+
+  depends_on :arm64
```
Comment (SMillerDev) [inline Formula/sse2neon.rb:9]: Missed a part here:

```suggestion
  depends_on arch: :arm64
```

**Review by SMillerDev — APPROVED** [review]
Thanks @alexreg  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**